addons/benchmark/benchmark.cc

```diff
@@ -856,27 +913,91 @@ int main (int argc, char *argv[]) {
 
       delete model;
     }
-    cout << endl;
   }
 
   if (benchmark_run_contacts) {
-    cout << "= Contacts: ForwardDynamicsConstraintsLagrangian" << endl;
-    contacts_benchmark (benchmark_sample_count, ContactsMethodLagrangian);
+    report_section("Contacts: ForwardDynamicsConstraintsDirect");
+    contacts_benchmark (benchmark_sample_count, ConstraintsMethodDirect);
 
-    cout << "= Contacts: ForwardDynamicsConstraintsRangeSpaceSparse" << endl;
-    contacts_benchmark (benchmark_sample_count, ContactsMethodRangeSpaceSparse);
+    report_section("Contacts: ForwardDynamicsConstraintsRangeSpaceSparse");
+    contacts_benchmark (benchmark_sample_count, ConstraintsMethodRangeSpaceSparse);
 
-    cout << "= Contacts: ForwardDynamicsConstraintsNullSpace" << endl;
-    contacts_benchmark (benchmark_sample_count, ContactsMethodNullSpace);
+    report_section("Contacts: ForwardDynamicsConstraintsNullSpace");
+    contacts_benchmark (benchmark_sample_count, ConstraintsMethodNullSpace);
 
-    cout << "= Contacts: ForwardDynamicsContactsKokkevis" << endl;
-    contacts_benchmark (benchmark_sample_count, ContactsMethodKokkevis);
+    report_section("Contacts: ForwardDynamicsContactsKokkevis");
+    contacts_benchmark (benchmark_sample_count, ConstraintsMethodKokkevis);
   }
 
   if (benchmark_run_ik) {
-    cout << "= Inverse Kinematics" << endl;
+    report_section("Inverse Kinematics");
     run_all_inverse_kinematics_benchmark(benchmark_sample_count);
   }
 
+  if (json_output) {
```

Comment:
While this code is functional correct. Please consider using a library (https://github.com/nlohmann/json#execute-unit-tests) for generating the json file or add some tests to prove that the output is correct json. Otherwise, we should consider refactoring the code for better maintainability.

Reply:
I don't think a distinct library is needed at this point. Writing json files is much simpler than parsing. Furthermore the number types are clear when writing. Should we export more complex stuff or rely on json more I would agree, though.